src/lib/io/sol-iio.c

```diff
@@ -1572,6 +1573,33 @@ resolve_absolute_path_cb(void *data, const char *dir_path, struct dirent *ent)
     return SOL_UTIL_ITERATE_DIR_CONTINUE;
 }
 
+static char *
+resolve_path(const char *path, char *resolved_path)
+{
+    int r;
+    glob_t result;
+
+    r = glob(path, GLOB_MARK | GLOB_TILDE, NULL, &result);
+    if (r != 0) {
+        switch (r) {
+            case GLOB_NOMATCH:
+                errno = ENOENT;
+                return NULL;
+            case GLOB_NOSPACE:
+                errno = ENOMEM;
+                return NULL;
+            case GLOB_ABORTED:
+            default:
+                errno = EINVAL;
+                return NULL;
+        }
+    }
+
+    resolved_path = realpath(result.gl_pathv[0], resolved_path);
```

Comment:
nope,  realpath() receives a pointer to an allocated memory and change this memory. But there is a problem in this code, sending a new pr. @fulong82 are you able to test it ?